### PR TITLE
Update RN version to 0.40

### PR DESCRIPTION
**Pull request by tafelito**
@kfiroo So in order to get the new keyboard props working, we need to update to the latest RN version. I managed to build GC with the latest RN making some changes on the xcode project. 

Here are all the changes we need to make it work with RN 0.40

I'm not sure how you want to handle this, I just create the PR, let me know if you think we should do something else.

This includes the changes in the info.plist to make the camera work as well

### CODE REVIEW

**Comment by tafelito** [comment]
we could create a new branch if you want, so we don't have to merge straight into master, and then it'll be easier to manage different release versions for GC

File: example/package.json [hunk 1]
```diff
@@ -7,8 +7,8 @@
     "sync": "rm -rf ./node_modules/react-native-gifted-chat; sane '/usr/bin/rsync -v -a --exclude .git --exclude example --exclude __tests__ --exclude node_modules ../ ./node_modules/react-native-gifted-chat/' .. --glob='{**/*.json,**/*.js}'"
   },
   "dependencies": {
-    "react": "15.2.1",
-    "react-native": "^0.30.0",
+    "react": "~15.4.0-rc.4",
```
Comment (kfiroo) [inline example/package.json]: Is this necessary?

Comment (tafelito) [inline example/package.json]: RN has the dependency on that specific react version. I tried to use the latest React version 15.4.2 but it does not work. I think when they release the latest RN to npm they'll update that too. I hope so

Comment (tafelito) [inline example/package.json]: you can see the peerDependency here [https://github.com/facebook/react-native/blob/master/package.json](https://github.com/facebook/react-native/blob/master/package.json)

Comment (sibelius) [inline example/package.json]: https://github.com/ncuillery/rn-diff can help as well

**Comment by kfiroo** [comment]
@tafelito This is only for the example project right?

**Comment by tafelito** [comment]
yes @kfiroo, only for the example project. GC does not have any hard dependency on RN

**Pull request merged**